File: lab/cpp/greet/greet5.cpp
#include <iostream> 
#include <string>

void greet(std::string &name)
{
	std::cout << "Hello, " + name + "!" << std::endl;
}

std::string greet_return_message(std::string &name)
{
	std::string message = "Hello, " + name + "!";
	return message;
}

int main()
{
	std::string name;
	std::cout << "What's your name? ";
	std::getline(std::cin, name);
	std::cout << "Hello, " + name + "!" << std::endl;
	std::string message;
	message = greet_return_message(name);
	std::cout << message << std::endl;
	greet(name);
}

File: lab/rust/greet/greet5.rs
use std::io;

fn greet(String )
{

}

fn main()
{
	let name : String;
	println!("What's your name? ");
	io::stdin().read_line(&mut name).expect("Failed to read line"):
	name = "Stefano".to_string();
	println!("Hello, {}!", name);
}
